api/cpp/tests/libraries/CMakeLists.txt

```
@@ -0,0 +1,11 @@
+# Copyright © SixtyFPS GmbH <info@slint.dev>
+# SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+add_executable(libraries main.cpp)
+target_link_libraries(libraries PRIVATE Slint::Slint)
+slint_target_sources(libraries appwindow.slint
+    LIBRARY_PATHS
+       helper_components ${CMAKE_CURRENT_SOURCE_DIR}/../../../../tests/helper_components/
+       helper_buttons ${CMAKE_CURRENT_SOURCE_DIR}/../../../../tests/helper_components/test_button.slint
```

Would it not be nicer if it was this syntax:

```suggestion
       helper_components=${CMAKE_CURRENT_SOURCE_DIR}/../../../../tests/helper_components/
       helper_buttons=${CMAKE_CURRENT_SOURCE_DIR}/../../../../tests/helper_components/test_button.slint
```

This would be consistant with the syntax in other language. But Maybe it's not idiomatic in cmake?

---

Usually in Cmake you would define a local variable for that:
`set(helper_buttons ...)` then use it as a variable wherever you want in the same file `${helper_buttons}`

---

As it is right now is more inclined towards Cmake standards. Usually you have a tuple (e.g. id and value) as separate parameters,  always in succession. An clear example would be the usage of `target_link_libraries` where you can specify the scope:
```
target_link_libraries(${PROJECT_NAME} 
    PRIVATE Slint::Slint
    PRIVATE pthread
)
```
Although in this case the scope is optional which raises the question, what happens if a name is not given to the library?

---

I'm a little torn. On the one hand `target_link_libraries` uses single value options here, so `LIBRARY_PATHS name1 lib1 name2 lib2` isn't quite strictly idiomatic cmake. On the other hand, I think `=` doesn't feel like CMake syntax either.

Looking around for more precedence.

---

Alternate syntax proposal:

```cmake
slint_target_sources(libraries appwindow.slint
    LIBRARY helper_components ${CMAKE_CURRENT_SOURCE_DIR}/../../../../tests/helper_components/
    LIBRARY helper_buttons ${CMAKE_CURRENT_SOURCE_DIR}/../../../../tests/helper_components/test_button.slint
)
```

---

@hunger what do you think? Can you think of any CMake API that can serve as inspiration?